crates/libcontainer/src/container/tenant_builder.rs

```diff
@@ -136,7 +150,13 @@ impl<'a> TenantContainerBuilder<'a> {
         let mut notify_socket = NotifySocket::new(notify_path);
         notify_socket.notify_container_start()?;
 
-        Ok(pid)
+        close(write_end)?;
+
+        let mut buf = [0; 1024];
```

Comment:
This implementation has a limit on the number of characters in error messages from init process, right? I would like to remove this. I have considered some ideas
- Repeat read
- structured message
    - Pass the length of the message first, separated by `:`, etc.
- Make two communications
    1. Sent the length of a message
    2. Sent the body of a message.

Reply:
Yes, I have also considered repeated reading until read returns 0, which will allow us to get complete message. Once rest of the code and approach is finalized and approved, I'll make that change.

I don't think we could use the third way, as far as know the data written to pipes is buffered, so the length and body will come out as a single data stream.

Reply:
I'm a little concerned when veeeeeeery long error message comes in

Reply:
@YJDoc2 If you need a lot of time to address it, please create an issue. and let's merge this PR ;)

Reply:
Hey, I was a bit busy for last couple of days, working on this now. Will probably get this done today or tomorrow. :)